File: src/Tawtheef.Websites/operations-web/src/app/pages/user/employee/test-slots-management/models/create-test-slot.dto.ts
export const TestSlotStaffRoleIds = {
  hallSupervisor: 'f8535d6c-a02e-46d8-a124-db986e51d5fc',
  monitor: '7020e361-a374-4ebd-ab69-dd95d1de8789',
} as const;

export type TestSlotStaffRoleId =
  (typeof TestSlotStaffRoleIds)[keyof typeof TestSlotStaffRoleIds];

export interface CreateTestSlotStaffDto {
  staffUserId: string;
  roleId: TestSlotStaffRoleId;
  isActive: boolean;
}

export interface CreateTestSlotDto {
  titleAr: string;
  titleEn: string;
  roomId: string;
  slotDate: string;
  startTime: string;
  endTime: string;
  staff: CreateTestSlotStaffDto[];
}

export interface UpdateTestSlotAssignmentsDto {
  staff: CreateTestSlotStaffDto[];
}

export interface TestSlotConfigurationDto extends Omit<CreateTestSlotDto, 'staff'> {
  id: string;
  staff: TestSlotConfigurationStaffDto[];
}

export interface TestSlotConfigurationStaffDto {
  staffUserId: string;
  name: string;
  roleId: string;
}

export interface TestSlotStaffConflictDto {
  staffUserId: string;
  staffDisplayName: string;
  slotTitle: string;
  slotDate: string;
  startTime: string;
  endTime: string;
}

File: src/Application.Operation/Features/Employee/TestSlots/DTOs/CreateTestSlotDto.cs
namespace Application.Operation.Features.Employee.TestSlots.DTOs;

public sealed class CreateTestSlotDto
{
    public required string TitleAr { get; init; }
    public string? TitleEn { get; init; }
    public Guid RoomId { get; init; }
    public DateOnly SlotDate { get; init; }
    public TimeOnly StartTime { get; init; }
    public TimeOnly EndTime { get; init; }
    public List<TestSlotStaffAssignmentDto>? Staff { get; init; }
}

public sealed class TestSlotStaffAssignmentDto
{
    public Guid StaffUserId { get; init; }
    public Guid RoleId { get; init; }
    public bool IsActive { get; init; }
}

public sealed class UpdateTestSlotAssignmentsDto
{
    public List<TestSlotStaffAssignmentDto>? Staff { get; init; }
}

public sealed record SavedTestSlotDto(Guid Id);

public sealed class TestSlotConfigurationDto
{
    public Guid Id { get; init; }
    public required string TitleAr { get; init; }
    public string? TitleEn { get; init; }
    public Guid RoomId { get; init; }
    public DateOnly SlotDate { get; init; }
    public TimeOnly StartTime { get; init; }
    public TimeOnly EndTime { get; init; }
    public required List<TestSlotConfigurationStaffDto> Staff { get; init; }
}

public sealed class TestSlotConfigurationStaffDto
{
    public Guid StaffUserId { get; init; }
    public required string Name { get; init; }
    public Guid RoleId { get; init; }
}

public sealed record TestSlotStaffConflictDto(
    Guid StaffUserId,
    string StaffDisplayName,
    string SlotTitle,
    DateOnly SlotDate,
    TimeOnly StartTime,
    TimeOnly EndTime);
